src/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -199,7 +199,22 @@ void CodegenNeuronCppVisitor::print_namespace_stop() {
 /// TODO: Edit for NEURON
 std::string CodegenNeuronCppVisitor::float_variable_name(const SymbolType& symbol,
                                                          bool use_instance) const {
-    return symbol->get_name();
+    auto name = symbol->get_name();
+    auto dimension = symbol->get_length();
+    // auto position = position_of_float_var(name);
+    // clang-format off
+    if (symbol->is_array()) {
+        if (use_instance) {
+            return fmt::format("(inst.{}+id*{})", name, dimension);
+        }
+        throw std::runtime_error("Not implemented.");
+        // return fmt::format("(data + {}*pnodecount + id*{})", position, dimension);
```

Comment:
It'll come back the very instant the throw goes away. The idea is first person to trigger the assert gets to check that there are proper tests, etc. I've already started on checking that ODEs with array range variables work. This is just a way of creating merge conflicts :)

Reply:
Okay. I would prefer having a TODO which would do the same thing instead of the commented code that isn't immediately understood why it's there but if you're working on that at the moment I'm fine with it